dashboard/src/apollo/client.js

```diff
@@ -27,34 +30,35 @@ const createClient = () => {
     dataIdFromObject: object => object.id,
   });
 
-  const authLink = setContext(() =>
-    getAccessToken().then(token => ({
-      headers: { Authorization: `Bearer ${token}` },
-    })),
-  );
-
-  const errorLink = onError(({ graphQLErrors, networkError }) => {
-    // TODO: Connect this error handler to display a blocking error alert
-    if (graphQLErrors)
-      graphQLErrors.forEach(error => {
-        // eslint-disable-next-line no-console
-        console.error(error.originalError || error);
-      });
-    // eslint-disable-next-line no-console
-    if (networkError) console.error(networkError);
-  });
-
-  const httpLink = new HttpLink({
-    uri: "/graphql",
-    fetchOptions: {},
-    credentials: "same-origin",
-  });
+  let client = null;
+  const getClient = () => {
+    if (!client) throw new Error("apollo client is not initialized");
+    return client;
+  };
 
-  const client = new ApolloClient({
+  client = new ApolloClient({
     cache,
-    link: ApolloLink.from([errorLink, authLink, httpLink]),
+    link: ApolloLink.from([
+      errorLink(),
+      stateLink({ cache }),
+      authLink({ getClient }),
+      httpLink(),
+    ]),
   });
 
+  localStorageSync(
+    client,
+    gql`
+      query SyncAuthQuery {
+        auth {
```

Comment:
Right, good catch. It should, but it still happens to work since the query is only used on the cache which makes no differentiation between client and remote.